File: python/problem_392_is_subsequence.py
# https://leetcode.com/problems/is-subsequence/

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i,j=0,0
        size_s = len(s)
        size_t = len(t)

        while i<size_s and j<size_t:
            if s[i]==t[j]:
                i=i+1
            j=j+1
        
        return i==size_s
        

File: go/problem392_is_subsequence.go
// https://leetcode.com/problems/is-subsequence

func isSubsequence(s string, t string) bool {
    sizeS := len(s)
    sizeT := len(t)

    if sizeS == 0 {
        return true
    }

    i,j := 0,0
    
    for i<sizeS && j<sizeT {
        if (s[i]==t[j]){
            i++
        }
        j++
    }

    return i==sizeS
}
